packages/loader/container-loader/src/container.ts

```diff
@@ -1061,8 +1062,11 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
         if (this.service === undefined) {
             throw new Error("Not attached");
         }
-        const storageService = await this.service.connectToStorage();
-        return new PrefetchDocumentStorageService(storageService);
+        let storageService = await this.service.connectToStorage();
+        if (this.service.servicePolicy?.serviceCachingImpl !== CachingPolicy.Prefetch) {
```

Comment:
In my original PR there was a feedback that it's confusing what names mean.
Does this look better:
```
if (this.service.policy?.caching !== CachingPolicy.DriverPrefetches) {
```

Or something like that, Basically CachingPolicy.Prefetch does not say who prefeteches, so we try to compensate it with serviceCachingImpl  thing that is too long and might be confusing.

Reply:
Even though the grammar would be wrong, this is essentially a boolean - so something like "IsPrefetch" would be better IMO.